File: src/src/main/java/com/itechart/sample/web/controller/LoginController.java
package com.itechart.sample.web.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.servlet.ModelAndView;

/**
 * @author andrei.samarou
 */
@Controller
public class LoginController {

    @RequestMapping(value = "/login", method = RequestMethod.GET)
    public ModelAndView login(@RequestParam(value = "error", required = false) String error) {
        ModelAndView model = new ModelAndView("login");
        if (error != null) {
            model.addObject("error", "Invalid username and password!");
        }
        return model;
    }

    @RequestMapping(value = "/logout", method = RequestMethod.GET)
    public ModelAndView logout() {
        ModelAndView model = new ModelAndView("login");
        model.addObject("error", "You are logged out successfully!");
        return model;
    }
}

File: src/security-web/src/main/webapp/app/login/login.controller.js
(function () {
    'use strict';

    angular
        .module('app')
        .controller('LoginController', LoginController);

    LoginController.$inject = ['$location', '$http', 'AuthService'];

    function LoginController($location, $http, AuthService) {
        var vm = this;
        vm.login = function () {
            AuthService.login(vm.username, vm.password).then(
                function (response) {
                    $location.path('/users');
                },
                function (error) {
                    console.log(error);
                    vm.error = "Invalid login or password";
                });
        };
    }

})();